File: backend/app/schemas/employee.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class EmployeeBase(BaseModel):
    employee_code: str
    full_name: str
    email: str
    department: str | None = None


class EmployeeCreate(EmployeeBase):
    user_id: int


class EmployeeUpdate(BaseModel):
    employee_code: str | None = None
    full_name: str | None = None
    email: str | None = None
    department: str | None = None


class EmployeeResponse(EmployeeBase):
    id: int
    user_id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/employee.ts
export interface Employee {

  id: number;

  employee_code: string;

  full_name: string;

  email: string;

  department: string | null;

  user_id: number;

  created_at?: string;

  updated_at?: string;
}


export interface EmployeeCreate {

  employee_code: string;

  full_name: string;

  email: string;

  department: string | null;

  user_id: number;
}


export interface EmployeeUpdate {

  employee_code?: string;

  full_name?: string;

  email?: string;

  department?: string | null;

  user_id?: number;
}